app/controllers/documents.js

```diff
@@ -0,0 +1,143 @@
+class Document{
+
+  constructor(){
+    this.jwt = require('jsonwebtoken');
+    this.config = require('./../config/config');
+    this.models = require('./../models/index');
+  }
+
+  createDoc(req, res){
+
+    const verifyToken = this.jwt.verify(req.headers.authorization, this.config.secrete);
+    this.models.documents.createDoc(req, verifyToken.id, verifyToken.role).then((document)=>{
+      res.status(201).json({
+        status: 'success',
+        message: 'Document created successfully',
+        data: document
+      });
+    }).catch((error)=>{
+      res.status(403).json({
+        status: 'fail',
+        message: error
+      })
+    });
+  }
+
+  all(req, res){
+    const
+      limit = req.query.limit,
+      offset = req.query.offset,
+      byRole = req.query.role,
+      byDate = req.query.date,
+      verifyToken = this.jwt.verify(req.headers.authorization, this.config.secrete);
+    this.models.documents.all(limit, offset, byRole, byDate, verifyToken.id, verifyToken.role)
+      .then((docs)=>{
+        res.status(200).json({
+          status: 'success',
+          message: 'Documents listed',
+          data: docs
+        })
+      }).catch((errorDetails)=>{
+      res.status(errorDetails.statusCode).json({
+        status: 'fail',
+        message: errorDetails.message
+      });
+    })
+  }
+
+  getDoc(req, res){
+    const docId = req.params.id;
+    const verifyToken = this.jwt.verify(req.headers.authorization, this.config.secrete);
+    this.models.documents.getDoc(verifyToken.id, docId)
+      .then((document)=>{
+        res.status(200).json({
+          status: 'success',
+          message: 'Document info loaded',
+          data: document
+        })
+      }).catch((errorDetails)=>{
+        res.status(errorDetails.statusCode).json({
+          status: 'fail',
+          message: errorDetails.message
+        });
+    })
+  }
+
+  updateDoc(req, res){
+    const verifyToken = this.jwt.verify(req.headers.authorization, this.config.secrete);
+    const docId = req.params.id;
+    const title = req.body.title;
+    const content = req.body.content;
+    const access = req.body.access;
+
+    const Validator = require('./../helpers/validate');
```

Comment:
You shouldn't require modules within other code as it makes it hard to find. Require this module in the constructor and assign to `this.validator` like you did for the others. A central reference point for all required modules is better